types/styled-components/test/index.tsx

```diff
@@ -714,6 +714,24 @@ async function typedThemes() {
         ${themedCssWithNesting}
     `;
 
+    const WithProp = styled.div`
+        ${({ ok, theme: { color } }: { ok: boolean; theme: typeof theme }) =>
+            ok &&
+            css`
+                color: ${color};
+            `}
+    `;
+
+    // TS 3.7 note: this breaks when FlattenInterpolation is an interface,
+    // it needs to be made a recursive type to fix.
+    const WithPropNested = styled.div`
+        ${({ ok }: { ok: boolean }) =>
+            ok &&
+            css`
+                color: ${({ theme: { color } }) => color};
```

Comment:
Fixing `FlattenInterpolation` (or rolling back typescript versions) also fixes this `--implicitAny` error. I only see it with the IDE service, though, I think the compiler output in the tests gives up while printing the outer errors.